### PR TITLE
simplify parse config and delete error.go

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
The current dumpling code logic is not clear.

### What is changed and how it works?
1. Refine dumpling parse config code logic. DM can simplify parsing `ExtraArguments`. https://github.com/pingcap/dm/blob/5003e431f07169377f9afa3739ed0dace54be2fb/dumpling/util.go#L55
2. Use `pingcap/errors` package to remove `error.go`.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 
### Release note

- No release note

### CODE REVIEW

**Comment by lichunzhu** [comment]
cc @lance6716

File: v4/export/config.go [hunk 1]
```diff
@@ -137,10 +470,11 @@ func (config *Config) createExternalStorage(ctx context.Context) (storage.Extern
...
 const (
-	UnspecifiedSize          = 0
-	DefaultTiDBMemQuotaQuery = 32 * (1 << 30)
-	DefaultStatementSize     = 1000000
-	TiDBMemQuotaQueryName    = "tidb_mem_quota_query"
+	UnspecifiedSize       = 0
+	DefaultStatementSize  = 1000000
+	TiDBMemQuotaQueryName = "tidb_mem_quota_query"
+
+	DefaultTiDBMemQuotaQuery uint64 = 32 * (1 << 30)
```
Comment (kennytm) [inline v4/export/config.go:477]:
```suggestion
	DefaultTiDBMemQuotaQuery uint64 = 32 << 30
```

**Review by kennytm — COMMENTED** [review]
rest LGTM. please resolve conflict.

File: v4/export/config.go (hunk 1 above)
Comment (lichunzhu) [reply v4/export/config.go]: addressed

**Review by kennytm — APPROVED** [review]

**Pull request merged**